pkg/config/loader_test.go

```diff
@@ -0,0 +1,305 @@
+package config
+
+import (
+	"os"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+
+	autoheal "github.com/openshift/autoheal/pkg/apis/autoheal/v1alpha2"
+	batch "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestFile(t *testing.T) {
+	l := NewLoader()
+	l.File("test_file")
+	if len(l.files) != 1 {
+		t.Fail()
+	}
+
+	if l.files[0] != "test_file" {
+		t.Fail()
+	}
+}
+
+func TestFiles(t *testing.T) {
+	l := NewLoader()
+	l.Files([]string{"test_file0", "test_file1", "test_file2"})
+	if len(l.files) != 3 {
+		t.Fail()
+	}
+
+	for idx, file_name := range l.files {
+		if file_name != ("test_file" + strconv.Itoa(idx)) {
+			t.Fail()
+		}
+	}
+}
+
+func TestLoadFile(t *testing.T) {
+	filename := "test_config.yml"
+	file, _ := os.Create(filename)
+
+	defer file.Close()
+	defer os.Remove(filename)
+
+	configsTest := []struct {
+		configString string
+		expected     *Config
+	}{
+		{
+			configString: "",
+			expected: &Config{
+				awx: &AWXConfig{
+					jobStatusCheckInterval: time.Duration(5) * time.Minute,
+				},
+				throttling: &ThrottlingConfig{
+					interval: time.Duration(1) * time.Hour,
+				},
+			},
+		},
+		{
+			configString: `awx:
```

Comment:
Can we indent this text? I mean something like this:

```go
configString: `
    awx:
      address: https://my-awx.example.com/api
      ...
`
```

Maybe we could add a helper function that writes the YAML text to a file, removing from all lines the white space prefix of the first line.